python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -782,17 +785,57 @@ def trace(model: nn.Module,
         invalidInputError(False, "Accelerator {} is invalid.".format(accelerator))
 
     @staticmethod
-    def get_context(model: nn.Module):
+    def get_context(model: nn.Module, *models):
         """
-        Obtain corresponding context manager from model, defaults to BaseContextManager().
+        Obtain corresponding context manager from (multi) model, defaults to BaseContextManager().
 
         :param model: Any model of torch.nn.Module, including all models accelareted by
                InferenceOptimizer.trace/InferenceOptimizer.quantize.
-        :return: a context manager.
+        :param models: Any model of torch.nn.Module or list of torch.nn.Module, including all models
+               accelareted by InferenceOptimizer.trace/InferenceOptimizer.quantize.
+        :return: a context manager if there is no conflict between context managers,
+                 otherwise will report RuntimeError.
         """
-        if hasattr(model, "context_manager"):
-            return model.context_manager
-        return generate_context_manager(accelerator=None, precision="fp32")
+        def obtain_manager(input_model):
+            if hasattr(input_model, "_nano_context_manager"):
+                _context_manager = input_model._nano_context_manager
+            else:
+                _context_manager = generate_context_manager(accelerator=None,
+                                                            precision="fp32")
+            return _context_manager
+
+        def join_manager(manager, new_manager):
+            def is_bf16(x):
+                return isinstance(x, AutocastContextManager)
+            if (is_bf16(manager) ^ is_bf16(new_manager)) is True:
+                invalidInputError(False, "Can't obtain a new context manager for BF16 model"
+                                  "and non BF16 model.")
+            thread_num1 = manager.thread_num
+            thread_num2 = new_manager.thread_num
+            if thread_num1 != thread_num2:
+                if thread_num1 is None or thread_num2 is None:
+                    warnings.warn("One of the two models has thread control and the other "
+                                  "does not. The returned context manager will be dominated "
+                                  "by the non-None one.")
+                    if thread_num1 is None:
+                        return new_manager
+                    else:
+                        return manager
+                else:
+                    invalidInputError(False, "Can't obtain a new context manager for model"
```

Comment:
Do we want to return error here? We can always return a "common denominator" (e.g., no_grad and autocast with proper thread_num) to the users to run their code.

I think it's better to report warning instead of throwing error to block the user.

Reply:
In this case, two context managers have different thread number, what should be the proper thread_num?
Or we just report warning and set thread_num=None?

Reply:
will change to this in next PR.